src/main/kotlin/lotto/domain/Lotto.kt

```diff
@@ -1,32 +1,24 @@
 package lotto.domain
 
-data class Lotto(val lotto: List<LottoBall>) {
+data class Lotto(val lotto: Set<LottoBall>) {
 
     init {
         require(lotto.size == LOTTO_SIZE) { "비정상적인 로또입니다." }
     }
 
-    fun matchNumberWith(winningLotto: Lotto): Int {
-        return this.lotto.map {
-            if (winningLotto.contains(it)) 1 else 0
-        }.sum()
-    }
-
-    private fun contains(lottoBall: LottoBall): Boolean {
-        lotto.find { it == lottoBall }?.let {
-            return true
-        }
-        return false
-    }
-
+    fun matchNumberWith(winningLotto: Lotto): Int = this.lotto.intersect(winningLotto.lotto.toSet()).size
```

Comment:
```suggestion
    fun matchNumberWith(winningLotto: Lotto): Int = this.lotto.intersect(winningLotto.lotto).size
```

toSet은 없어도 될 것 같습니다!